### ISSUE TITLE
New error objects in rails 6.1

### ISSUE THREAD

**Issue by cherbst-2112** [trailblazer/reform#524]
### Complete Description of Issue

Hi, I'm trying to upgrade my reform 2.3 app to rails 6.1 and seeing a few issues. I found https://github.com/trailblazer/reform-rails/issues/86 but did not see any updates so thought I'd ask here if anyone had any ideas.

Rails 6.1 has a new errors object so they should use the new dsl, they have a few convenience methods to maintain compatibility but they are not working well with reform, specificaully `form.errors[:some_field]` if I want to see the errors on some field (this is nil).

### Steps to reproduce

I reproduced the problem in a new demo rails 6.1 app, here are the relevant parts:

```
ActiveRecord::Schema.define(version: 2021_01_12_190713) do

  create_table "things", force: :cascade do |t|
    t.string "name"
    t.datetime "created_at", precision: 6, null: false
    t.datetime "updated_at", precision: 6, null: false
  end

end
```

```
class ThingForm < Reform::Form
  property :name
  validates :name, presence: true
end
```

```
class ThingsController < ApplicationController
...
  # POST /things
  # POST /things.json
  def create
    @form = ThingForm.new(Thing.new)

    respond_to do |format|
      # HERE, if name is NULL, there is no error indexed with the field name, but there is an error present.
      if @form.validate(thing_params)
        @form.save

        format.html { redirect_to @thing, notice: 'Thing was successfully created.' }
        format.json { render :show, status: :created, location: @thing }
      else
        format.html { render :new }
        format.json { render json: @thing.errors, status: :unprocessable_entity }
      end
    end
  end
```

### Expected behavior

A normal active model validation will have an error:

```
(byebug) @form.model.errors[:name]
["can't be blank"]
```

### Actual behavior

The form will not have an error for name:

```
(byebug) thing_params
#<ActionController::Parameters {"name"=>""} permitted: true>
(byebug) @form.validate(thing_params)
false
(byebug) @form.errors[:name]
[]
```

Some methods *will* work correctly so it is a little confusing

```
(byebug) @form.errors.full_messages
["Name can't be blank"]
```

### System configuration
**Reform version**:
reform 2.3.3
reform-rails 0.2.1

**Comment by jamesbebbington** [comment]
Yes, I can confirm seeing the same issue when attempting to upgrade to rails 6.1. [`@dotted_errors`](https://github.com/trailblazer/reform/blob/bcae40b7405a9ac3ee46507fa2c85acbc997a8c0/lib/reform/errors.rb#L7-L22) in `Reform::Contract::Result::Errors` doesn't seem to be being set, not sure why though.

**Comment by marcelolx** [comment]
@cherbst-2112 @jamesbebbington As I mentioned here https://github.com/trailblazer/reform-rails/issues/86#issuecomment-758872874, I think the best approach to fix this is `reform-rails` to have its own implementation of `filter_for`, or send down to reform the data as reform expects.

Anyway, this issue is related to `reform-rails` and not `reform`, lets keep track of this issue here https://github.com/trailblazer/reform-rails/issues/86

**Comment by skcc321** [comment]
Guys you can use dry validation stand-alone. It has everything what you need for this case

**Comment by cherbst-2112** [comment]
And includes it's own DSL LOL. I considered porting all my validations to dry-validations instead but reworking them them to the new DSL was a heavier lift than fixing this library.

**Comment by apotonick** [comment]
Hi there, I'm working on Reform 3.0 and  I wonder if there's any place where I can find a complete definition of the `errors` API in Rails, or maybe even you you folks use it. I know there's
1. `errors[:title] # local form's property`
2. `errors.messages` #  hash of local errors

but I know there's more, like the dotted errors for nested forms, etc. Can anyone point me to better docs or simply paste a few more usage examples? :beers:

**Comment by marcelolx** [comment]
@apotonick I think that is this what you're looking for 

https://api.rubyonrails.org/classes/ActiveModel/Errors.html 
https://api.rubyonrails.org/classes/ActiveModel/Error.html

**Comment by apotonick** [comment]
@marcelolx :100: Thanks! :heart: Checked it out yesterday. I remember something like those dotted errors in nested setups - do you know anything about that? Also, what if you have a nested form and you do `form.errors[:nested_form_name]`?

I am making massive progress with Reform 3 and will need testers very very soon (like by the end of the week or so). Anyone? It will be in `Trailblazer::Reform` namespace so you can use both versions.

**Comment by marcelolx** [comment]
@apotonick I'm not aware of those dotted errors, as far as I can remember in my prior job for nested forms we used `model.errors[:nested_model]`
